datatypes/src/main/java/org/hyperledger/besu/datatypes/Transaction.java

```diff
@@ -183,4 +187,126 @@ default Optional<? extends Quantity> getMaxFeePerBlobGas() {
+   *
+   * @return {@code true} if this is a contract-creation transaction; otherwise {@code false}
+   */
+  boolean isContractCreation();
+
+  /**
+   * Return the maximum fee per gas the sender is willing to pay for this transaction.
+   *
+   * @return max fee per gas in wei
+   */
+  Wei getMaxGasPrice();
+
+  /**
+   * Return the effective priority fee per gas for this transaction.
+   *
+   * @param maybeBaseFee base fee in case of EIP-1559 transaction
+   * @return priority fee per gas in wei
+   */
+  Wei getEffectivePriorityFeePerGas(final Optional<Wei> maybeBaseFee);
+
+  /**
+   * Return the versioned hashes for this transaction.
+   *
+   * @return optional list of versioned hashes
+   */
+  Optional<List<VersionedHash>> getVersionedHashes();
+
+  /**
+   * Return the blobs with commitments for this transaction.
+   *
+   * @return optional blobs with commitments
+   */
+  Optional<BlobsWithCommitments> getBlobsWithCommitments();
+
+  /**
+   * Return the address of the contract, if the transaction creates one
+   *
+   * @return address of new contract or empty otherwise
+   */
+  Optional<Address> contractAddress();
+
+  /**
+   * Return the access list in case of EIP-2930 transaction
+   *
+   * @return optional access list
+   */
+  Optional<List<AccessListEntry>> getAccessList();
+
+  /**
+   * Writes the transaction to RLP
+   *
+   * @param out the output to write the transaction to
+   */
+  void writeTo(final RLPOutput out);
+
+  /**
+   * Calculates the up-front cost for the gas and data gas the transaction can use.
+   *
+   * @param gasPrice the gas price to use
+   * @param blobGasPrice the data gas price to use
+   * @param totalBlobGas the gas the transaction can use
+   * @return the up-front cost for the gas the transaction can use.
+   */
+  Wei getUpfrontGasCost(final Wei gasPrice, final Wei blobGasPrice, final long totalBlobGas);
+
+  /**
+   * Calculates the up-front cost for the transaction.
+   *
+   * <p>The up-front cost is paid by the sender account before the transaction is executed. The
+   * sender must have the amount in its account balance to execute and some of this amount may be
+   * refunded after the transaction has executed.
+   *
+   * @param totalBlobGas the gas the transaction can use
+   * @return the up-front gas cost for the transaction
+   */
+  Wei getUpfrontCost(final long totalBlobGas);
+
+  /**
+   * Calculates the effectiveGasPrice of a transaction on the basis of an {@code Optional<Long>}
+   * baseFee and handles unwrapping Optional fee parameters. If baseFee is present, effective gas is
+   * calculated as:
+   *
+   * <p>min((baseFeePerGas + maxPriorityFeePerGas), maxFeePerGas)
+   *
+   * <p>Otherwise, return gasPrice for legacy transactions.
+   *
+   * @param baseFeePerGas optional baseFee from the block header, if we are post-london
+   * @return the effective gas price.
+   */
+  Wei getEffectiveGasPrice(final Optional<Wei> baseFeePerGas);
```

Comment:
This can be derived from fields already in the transaction interface.

Reply:
Removed